### PR TITLE
[CLOUDCRAFT-388] Automate Snapshots of Cloud Accounts via the Cloudcraft API

### PR DESCRIPTION
**Pull request by jamesponddotco**
### What does this PR do? What is the motivation?
Add a new article to the Cloudcraft documentation describing how to mimic the web UI's **Auto Layout** feature using the developer API and simple command-line tools.

Not really sure I like the title. Suggestions are welcome!

### Merge instructions
Please wait until the Cloudcraft team has reviewed the content before going ahead with the review and merging of the article.

- [x] Please merge after reviewing

### CODE REVIEW

File: content/en/cloudcraft/advanced/auto-layout-via-api.md
Comment (srividya-rajagopalan) [inline content/en/cloudcraft/advanced/auto-layout-via-api.md:28]: Instead of calling out separately for AWS and Azure , you can also make that a parameter like PROVIDER where provider value is `aws` or `azure`

File: content/en/cloudcraft/advanced/auto-layout-via-api.md
Comment (srividya-rajagopalan) [inline content/en/cloudcraft/advanced/auto-layout-via-api.md:54]: Instead of calling out separately for AWS and Azure , you can also make that a parameter like PROVIDER where provider value is `aws` or `azure`

File: content/en/cloudcraft/advanced/auto-layout-via-api.md
Comment (srividya-rajagopalan) [inline content/en/cloudcraft/advanced/auto-layout-via-api.md:22]: Do we need to add a step here to get the AWS / Azure account id using list accounts API and identify the right value for ACCOUNT_ID?

Since these ACCOUNT_IDs generated by Cloudcraft, the user will not know this offhand.

**Review by srividya-rajagopalan — CHANGES_REQUESTED** [review]

File: content/en/cloudcraft/advanced/auto-layout-via-api.md
Comment (jamesponddotco) [reply content/en/cloudcraft/advanced/auto-layout-via-api.md:22]: I think that should be its own article, which we need anyway. This come up on support all the time because of access control.

File: content/en/cloudcraft/advanced/auto-layout-via-api.md
Comment (jamesponddotco) [reply content/en/cloudcraft/advanced/auto-layout-via-api.md:28]: Oh, great idea! Doing that now.

File: content/en/cloudcraft/advanced/auto-layout-via-api.md
Comment (jamesponddotco) [reply content/en/cloudcraft/advanced/auto-layout-via-api.md]: Fixed with [8cb0721](https://github.com/DataDog/documentation/pull/22262/commits/8cb0721724ea3ad5c892c35119d94c85f74c619f).

File: content/en/cloudcraft/advanced/auto-layout-via-api.md
Comment (jamesponddotco) [reply content/en/cloudcraft/advanced/auto-layout-via-api.md]: Fixed with [685aacc](https://github.com/DataDog/documentation/pull/22262/commits/685aacc3b5f5d89410ea3802276ae65ce1f19b4a).

File: content/en/cloudcraft/advanced/auto-layout-via-api.md
Comment (jamesponddotco) [reply content/en/cloudcraft/advanced/auto-layout-via-api.md]: Fixed with [685aacc](https://github.com/DataDog/documentation/pull/22262/commits/685aacc3b5f5d89410ea3802276ae65ce1f19b4a).

File: content/en/cloudcraft/advanced/auto-layout-via-api.md
Comment (srividya-rajagopalan) [reply content/en/cloudcraft/advanced/auto-layout-via-api.md:22]: Should we alteast then mention that in the pre-requisites - you should have the Cloudcraft issued identifier for the interested AWS account handy?

**Comment by drichards-87** [comment]
@jamesponddotco I'll handle the review for the Documentation Team, so let me know when the PR is ready. Thanks!

File: content/en/cloudcraft/advanced/auto-layout-via-api.md
Comment (jamesponddotco) [reply content/en/cloudcraft/advanced/auto-layout-via-api.md:23]: Added on [c672e98](https://github.com/DataDog/documentation/pull/22262/commits/c672e98c3afa08a4c355494a46ef046643ef83cc).

File: content/en/cloudcraft/advanced/auto-layout-via-api.md
Comment (drichards-87) [inline content/en/cloudcraft/advanced/auto-layout-via-api.md:38]:
```suggestion
Replace `PROVIDER` with the cloud provider, for example, `azure` or `aws`, `ACCOUNT_ID` with the ID of your AWS or Azure account in Cloudcraft, `REGION` with your desired scan region, and `API_KEY` with your Cloudcraft API key.
```

File: content/en/cloudcraft/advanced/auto-layout-via-api.md
Comment (drichards-87) [inline content/en/cloudcraft/advanced/auto-layout-via-api.md:40]:
```suggestion
After executing the command, the JSON representation of your AWS account snapshot is displayed. To save this output directly to a file, use the following command:
```

File: content/en/cloudcraft/advanced/auto-layout-via-api.md
Comment (drichards-87) [inline content/en/cloudcraft/advanced/auto-layout-via-api.md:71]:
```suggestion
Remember to replace `API_KEY` with your actual Cloudcraft API key.
```

File: content/en/cloudcraft/advanced/auto-layout-via-api.md
Comment (drichards-87) [inline content/en/cloudcraft/advanced/auto-layout-via-api.md:55]:
```suggestion
Next, create a new blueprint in your Cloudcraft account using the [Create blueprint][4] API endpoint. The saved snapshot data serves as the payload for this request.
```

File: content/en/cloudcraft/advanced/auto-layout-via-api.md
Comment (drichards-87) [inline content/en/cloudcraft/advanced/auto-layout-via-api.md:51]:
```suggestion
The snapshot is saved with the filename `account-infra.json` in your temporary directory.
```

File: content/en/cloudcraft/advanced/auto-layout-via-api.md
Comment (drichards-87) [inline content/en/cloudcraft/advanced/auto-layout-via-api.md:75]:
```suggestion
If you have any questions or trouble with the process, [contact Cloudcraft's support team][5].
```

File: content/en/cloudcraft/advanced/auto-layout-via-api.md
Comment (drichards-87) [inline content/en/cloudcraft/advanced/auto-layout-via-api.md:73]:
```suggestion
Upon completion, a new blueprint reflecting your cloud infrastructure is created in your Cloudcraft account, replicating the effect of manually using the **Scan Now** and **Auto Layout** buttons.
```

**Review by drichards-87 — APPROVED** [review]
Left you some light feedback and approved the PR.

**Comment by drichards-87** [comment]
@jamesponddotco Just double-checking - are we still waiting for another approval, or are we good to merge the PR?

**Comment by jamesponddotco** [comment]
@drichards-87 We're good to merge :)

**Pull request merged**
